Feature/auth setup

FragaGabriel23 opened the pull request.

FragaGabriel23 committed:
feat(auth): Basic RBAC implementation

FragaGabriel23 committed:
feat(auth): implement middleware to handle Firebase Google login authentication

silveiratalita commented on src/modules/promotions/controllers/promotions.controller.ts:
```diff
@@ -10,29 +10,35 @@ import {
 export class PromotionsController {
   constructor(private readonly promotionsService: PromotionsService) {}
 
+  @Roles(Role.Admin, Role.Guest, Role.Store, Role.User)
```
Ajustar as roles, sendo que user é a entidade.

silveiratalita commented on src/modules/promotions/controllers/promotions.controller.ts:
```diff
@@ -10,29 +10,35 @@ import {
     return promotions;
   }
 
+  @Roles(Role.Admin, Role.Guest, Role.Store, Role.User)
```
Essa verificação deve mesmo ser feita no controller?

FragaGabriel23 committed:
refactor(auth): migrate from Firebase Realtime Database to PostgreSQL

- Replaced Firebase Realtime Database with PostgreSQL for user data storage
- Updated authentication and user retrieval logic
- Configured Firebase credentials through .env with the use of firebase-config.ts
- Implemented 'findOneUserById' route to fetch user by ID

FragaGabriel23 committed:
fix: update 'roles' column to accept single enum value with default role

FragaGabriel23 committed:
feat: add role assignment in middleware for user creation from JWT

FragaGabriel23 committed:
chore: add Firebase configuration keys to env.example

FragaGabriel23 committed:
feat(firebase): refactor authentication into a dedicated module

Refactored Firebase authentication by moving it from middleware to a dedicated Firebase module.
This improves code modularity and maintainability.
